library/src/hooks/places-service.ts

```diff
@@ -1,28 +1,49 @@
-import {useContext, useMemo} from 'react';
+import {useContext, useEffect, useState} from 'react';
 
 import {GoogleMapsContext} from '../google-maps-provider';
 
+export interface PlacesServiceProps {
+  divElement?: HTMLDivElement | null;
+}
+
 /**
  * Hook to get Google Maps Places Service instance
  */
-export const usePlacesService = (): google.maps.places.PlacesService | null => {
-  const {map} = useContext(GoogleMapsContext);
+export const usePlacesService = (
+  props?: PlacesServiceProps
+): google.maps.places.PlacesService | null => {
+  const {map, googleMapsAPIIsLoaded} = useContext(GoogleMapsContext);
 
-  // Creates a Places Service instance
-  const placesService = useMemo<google.maps.places.PlacesService | null>(() => {
-    // Wait for map to be initialized
-    if (!map) {
-      return null;
-    }
+  const [placesService, setPlacesService] =
+    useState<google.maps.places.PlacesService | null>(null);
 
-    if (!google.maps.places) {
+  // Creates a Places Service instance
+  useEffect(() => {
+    if (googleMapsAPIIsLoaded && !google.maps.places) {
       throw Error(
         "Places library missing. Add 'places' to the libraries array of GoogleMapsProvider."
       );
     }
 
-    return new google.maps.places.PlacesService(map);
-  }, [map]);
+    if (!props?.divElement) {
```

Comment:
I would check for `undefined`, in case someone wants to add a div element, but the element is still `null` (for example because it's not rendered yet) and there's a map in the application which shouldn't be used for the places service.
```suggestion
    if (props?.divElement === undefined) {
```

Reply:
Hm, good point! Needed a minute to understand the scenario.. but it is actually a thing. 👍